code/datums/movement/mob.dm

```diff
@@ -178,7 +178,9 @@
 	//We are now going to move
 	mob.moving = 1
 
-	direction = mob.AdjustMovementDirection(direction)
+	if (mob == mover)
+		direction = mob.AdjustMovementDirection(direction)
```

Comment:
I think the check needs to be inside this proc, not outside it--it's called from several places and might plausibly contain other logic. That might mean you need to pass mover as an arg to the proc.

Reply:
Yes, you are right. I've made adjustments